pkg/daemon/pivot/utils/run.go

```diff
@@ -48,9 +48,9 @@ func runExtBackoff(backoff wait.Backoff, command string, args ...string) (string
 		return true, nil
 	}); err != nil {
 		if err == wait.ErrWaitTimeout {
-			return "", errors.Wrapf(lastErr, "failed to run command %s (%d tries): %v", command, backoff.Steps, err)
+			return "", fmt.Errorf("failed to run command %s (%d tries): %v: %w", command, backoff.Steps, err, lastErr)
```

Comment:
Ahh good catch! This should probably be a two nested errors.

Reply:
Reasoning about this particular case, lets use an [aggregated error](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/errors/errors.go#L46) to represent the combination of `err` and `lastErr`. It should become something like this:

```golang
import (
  kubeErrs "k8s.io/apimachinery/pkg/util/errors"
)

// ...

errs := kubeErrs.NewAggregate([]error{err, lastErr})
return "", fmt.Errorf("failed to run command %s (%d tries): %v: %w", command, backoff.Steps, errs)
```

You shouldn't have to do anything to use that package because the MCO already uses it. See: https://github.com/openshift/machine-config-operator/blob/master/test/e2e-shared-tests/helpers.go#L219-L222 for an example.

Reply:
Thanks Zack!  

@raisaat I think there are a few more following this pattern in the files in this PR

Reply:
@cheesesashimi I can aggregate the errors using `kubeErrs` in [pkg/controller/node/node_controller.go](https://github.com/openshift/machine-config-operator/pull/2868/commits/8a7c5b44e5f4567a3d2b872488db3c571f25cc34#diff-92f10e93215b8da12723d913e0985568114aab4b8afb67541c2a3a9ec5cafba6L775-R794), [pkg/daemon/drain.go](https://github.com/openshift/machine-config-operator/pull/2868/commits/8a7c5b44e5f4567a3d2b872488db3c571f25cc34#diff-58db213cf6f6259a621f8d3421b5e8af55ed71c9f52c15b255f66f0b1e78a946R69), [pkg/daemon/update.go](https://github.com/openshift/machine-config-operator/pull/2868/commits/8a7c5b44e5f4567a3d2b872488db3c571f25cc34#diff-349c0748c3f52201852d3027c29daf618b45300b949482728df6666a3f9ba245), [pkg/operator/sync.go](https://github.com/openshift/machine-config-operator/pull/2868/commits/8a7c5b44e5f4567a3d2b872488db3c571f25cc34#diff-3e205577bec1a1d1711df8bfeff30e63f044b24cfbdc69bbd7d0052fdc881891R655-R665) and [test/e2e-single-node/sno_mcd_test.go](https://github.com/openshift/machine-config-operator/pull/2868/commits/8a7c5b44e5f4567a3d2b872488db3c571f25cc34#diff-8771ad0c06d4ca41d8201443cd09bc24541cedd0176bd013ced9c0e271b958feR408-R410) too right?

Reply:
Yes, you can do the same thing in those places as well. I never realized how often we used the pattern of having potentially multiple errors being returned! Anyway, by using the Aggregate error type, they'll appear to be a single error for checking purposes (e.g., `if err != nil { //... }`), but the individual errors will remain separate and can be unwrapped later, if necessary.